### PR TITLE
Merge 'local' and 'remote' parts of TimelineInfo into one struct.

### PR DESCRIPTION
**Pull request by hlinnaka**
This comes in two commits, for easier review. See commit messages for details

This is extracted from PR #2595.

I haven't run the full test suite on this yet, so I'm sure there are still failures, that's why I'm still keeping this as Draft. Other than that, this is ready for review.

### CODE REVIEW

File: control_plane/src/bin/neon_local.rs [hunk 1]
```diff
@@ -801,7 +775,7 @@ fn handle_pg(pg_match: &ArgMatches, env: &local_env::LocalEnv) -> Result<()> {
                         // Use the LSN at the end of the timeline.
                         timeline_infos
                             .get(&node.timeline_id)
-                            .and_then(|bi| bi.local.as_ref().map(|l| l.last_record_lsn.to_string()))
+                            .and_then(|bi| Some(bi.last_record_lsn.to_string()))
```
Comment (SomeoneToIgnore) [inline control_plane/src/bin/neon_local.rs:778]:
```suggestion
                            .map(|bi| bi.last_record_lsn.to_string())
```

**Review by SomeoneToIgnore — COMMENTED** [review]
Looks good generally, let's fix the tests first and approve after.

**Comment by hlinnaka** [comment]
Ok, all the test failures should be fixed now.

File: libs/pageserver_api/src/models.rs [hunk 2]
```diff
@@ -123,9 +123,15 @@ pub struct TenantInfo {
     pub has_in_progress_downloads: Option<bool>,
 }
 
+/// This represents the output of the "timeline_detail" and "timeline_list" API calls.
 #[serde_as]
 #[derive(Debug, Serialize, Deserialize, Clone)]
-pub struct LocalTimelineInfo {
+pub struct TimelineInfo {
+    #[serde_as(as = "DisplayFromStr")]
+    pub tenant_id: TenantId,
```
Comment (SomeoneToIgnore) [inline libs/pageserver_api/src/models.rs:131]: That would require changing https://github.com/neondatabase/neon/blob/9fe4548e13774d7f1e5f9b5d23e57da971419442/pageserver/src/http/openapi_spec.yml#L610-L625 and https://github.com/neondatabase/cloud/blob/a90901d4269b3fc1cbec1286f197ee07e70923f0/swagger/pageserver.yml#L558-L573

And the latter would affect the console:

https://github.com/neondatabase/cloud/blob/a90901d4269b3fc1cbec1286f197ee07e70923f0/goapp/internal/client/psclient/psclient.go#L86-L91

https://github.com/neondatabase/cloud/blob/a90901d4269b3fc1cbec1286f197ee07e70923f0/goapp/internal/operationsv2/provisioner_common.go#L588-L603

I think we can incorporate the changes in forward-compatible way, but need to discuss it further maybe.

File: pageserver/src/http/routes.rs [hunk 3]
```diff
@@ -567,36 +521,32 @@ async fn tenant_status(request: Request<Body>) -> Result<Response<Body>, ApiErro
             false
         });
 
-    let tenant_state = match tenant {
-        Ok(tenant) => tenant.current_state(),
+    let (tenant_state, current_physical_size) = match tenant {
+        Ok(tenant) => {
+            let timelines = tenant.list_timelines();
+            let current_physical_size = tokio::task::spawn_blocking(move || {
```
Comment (SomeoneToIgnore) [inline pageserver/src/http/routes.rs:527]: Looks like we can omit this `spawn_blocking` entirely? I see no blocking operations in the `get_physical_size` method (prometheus metrics are based on atomics).

https://github.com/neondatabase/neon/blob/e5e40a31f4ce84a37f88aad2fe612d144fc55e6f/pageserver/src/tenant/timeline.rs#L386-L388

File: test_runner/regress/test_timeline_size.py [hunk 4]
```diff
@@ -237,6 +233,9 @@ def test_timeline_physical_size_init(neon_simple_env: NeonEnv):
     env.pageserver.stop()
     env.pageserver.start()
 
+    # FIXME: wait a little, so that the tenant is loaded
```
Comment (SomeoneToIgnore) [inline test_runner/regress/test_timeline_size.py:236]: Can we check similar `has_in_progress_downloads` status thing?

File: test_runner/regress/test_timeline_size.py [hunk 5]
```diff
@@ -450,28 +449,30 @@ def assert_physical_size(env: NeonEnv, tenant_id: TenantId, timeline_id: Timelin
...
-    )
-    assert res["local"]["current_physical_size"] == get_timeline_dir_size(timeline_path)
+    assert res["current_physical_size"] == res["current_physical_size_non_incremental"]
+    assert res["current_physical_size"] == get_timeline_dir_size(timeline_path)
 
 
 # Timeline logical size initialization is an asynchronous background task that runs once,
 # try a few times to ensure it's activated properly
+#
+# FIXME: unused and broken now that we do on-demand download
```
Comment (SomeoneToIgnore) [inline test_runner/regress/test_timeline_size.py:461]: Not sure what that means, we use it in 3 placed of this file.

**Review by SomeoneToIgnore — COMMENTED** [review]
I really love splitting PRs, apparently, thanks to that I can think more on them.

Today, I've discovered that we're making breaking changes with it: nothing big, but some go code needs adjustments and nice that we've found it this way.
Actually, e2e tests fail too, now it's more visible.

File: libs/pageserver_api/src/models.rs (hunk 2 above)
Comment (hlinnaka) [reply libs/pageserver_api/src/models.rs:131]: Good catch, I didn't realize the control plane uses these. Obvious in hindsight..

I kept little stub versions of 'local' and 'remote' fields, for backwards-compatibility, that duplicate the fields that the control plane uses. We can remove them later, when the control plane has been updated. (But I don't want to update the control plane until the new pageserver version has been deployed all the way to production, and we're sure we don't need to roll back anymore).

File: test_runner/regress/test_timeline_size.py (hunk 4 above)
Comment (hlinnaka) [reply test_runner/regress/test_timeline_size.py]: This isn't actually needed at all. Not yet, anyway; it's needed with PR #2595.

**Review by SomeoneToIgnore — APPROVED** [review]
LGTM

Let's ensure e2e tests pass before the merge

File: pageserver/src/http/routes.rs (hunk 3 above)
Comment (hlinnaka) [reply pageserver/src/http/routes.rs]: Fixed, thanks!

**Pull request merged**